src/manage_sql_agent_groups.c

```diff
@@ -436,9 +436,26 @@ delete_agent_group (const char *agent_group_uuid, int ultimate)
            " SELECT %llu, agent_id FROM agent_group_agents WHERE group_id = %llu;",
            trash_id, agent_group);
 
+    /* Update the location of the agent_group in any trashcan tasks. */
+    sql ("UPDATE tasks"
+         " SET agent_group = %llu,"
+         "     agent_group_location = " G_STRINGIFY (LOCATION_TRASH)
+         " WHERE agent_group = %llu"
+         " AND agent_group_location = " G_STRINGIFY (LOCATION_TABLE) ";",
+         trash_id,
+         agent_group);
+
       permissions_set_locations ("agent_group", agent_group, trash_id, LOCATION_TRASH);
       tags_set_locations ("agent_group", agent_group, trash_id, LOCATION_TRASH);
     }
+  else if (sql_int ("SELECT count(*) FROM tasks"
```

Comment:
I think this should use the `agent_group_in_use` function to ensure the indicator that the agent group is still in use is consistent with the delete command failing because the agent group is still in use.